### PR TITLE
Refactor of Messages and MessageHandlers

### PR DESCRIPTION
**Pull request by Ranray**

### CODE REVIEW

File: Soccar [Game-Server]/src/nl/soccar/gameserver/message/PlayerLeaveSessionMessage.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * @author PTS34A
+ */
+@MessageEvent(id = MessageConstants.PLAYER_LEAVE_SESSION_MESSAGE_ID, handler = PlayerLeaveSessionMessageHandler.class)
+public class PlayerLeaveSessionMessage extends Message {
+
+    private final String username;
+    private final TeamColour colour;
+
+    public PlayerLeaveSessionMessage(String username, TeamColour colour) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/message/PlayerLeaveSessionMessage.java:19]: MINOR Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/SessionController.java [hunk 2]
```diff
@@ -130,21 +130,28 @@ private void sendJoinedSessionNotification(Player joinedPlayer, TeamColour joine
...
         player.setCurrentSession(null);
 
-        PlayerLeftSessionMessage m = new PlayerLeftSessionMessage(player.getUsername(), team.getTeamColour());
+        PlayerLeftSessionMessage m = new PlayerLeftSessionMessage(player.getUsername(), colour);
         room.getAllPlayers().stream().map(server::getConnectionFromPlayer).forEach(c -> c.send(m));
+        LOGGER.log(Level.INFO, "Player {0} left the Session.", player.getUsername());
         return true;
     }
+    
+    public void leaveSession(Player player, Session session) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/SessionController.java:147]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

**Comment by ghost** [comment]
SonarQube analysis reported 2 issues
* MINOR 2 minor

Watch the comments in this conversation to review them.

**Review by BeeHiveJava — APPROVED** [review]

**Comment by BeeHiveJava** [comment]
retest this please

**Pull request merged**